perlin_noise/perlin_noise.py

```diff
@@ -166,11 +167,20 @@ def perlin_noise(
     if ow != bw * gw != 0:
         raise Exception(f"Output width {ow} must be divisible by grid width {gw}")
 
-    angle = torch.empty(
-        [batch_size] + [s + 1 for s in grid_shape], *args, **kwargs
```

Comment:
Changing this line to `[batch_size or 1] + [s + 1 for s in grid_shape], *args, **kwargs` should be enough?